index.js

```diff
@@ -1,28 +1,43 @@
 'use strict';
 const net = require('net');
 
-const getPort = options => new Promise((resolve, reject) => {
-	// For backwards compatibility with number-only input
-	// TODO: Remove this in the next major version
-	if (typeof options === 'number') {
-		options = {
-			port: options
-		};
-	}
-
+const isAvailable = options => new Promise((resolve, reject) => {
 	const server = net.createServer();
-
 	server.unref();
 	server.on('error', reject);
-
 	server.listen(options, () => {
-		const port = server.address().port;
+		const {port} = server.address();
 		server.close(() => {
 			resolve(port);
 		});
 	});
 });
 
+const getPort = options => new Promise((resolve, reject) => {
+	// For backwards compatibility with number-only input
+	// TODO: Remove this in the next major version
+	if (typeof options === 'number') {
+		options = getOptions(options);
+	}
+
+	if (typeof options.port === 'number') {
+		options.port = [options.port];
+	}
+
+	options.port.reduce((seq, port) => {
+		return seq.catch(() => {
+			const input = getOptions(port, options.host);
+			return isAvailable(input)
+				.then(port => port)
+				.catch(Promise.reject.bind(Promise));
+		});
+	}, Promise.reject()).then(resolve).catch(reject);
+});
+
+function getOptions(port, host) {
```

Comment:
Don't you think this function can be removed after all?

The following will do just fine:

```
if (typeof options === 'number') {
   options = { port: options };
}
```

ànd

```
isAvailable(Object.assign({}, options, { port }));
```

With `Object.assign`, you retain the previously implemented behavior which is the ability to eventually pass more options to `server.listen`.